src/utils.ts

```diff
@@ -0,0 +1,90 @@
+import * as vscode from 'vscode';
+
+export async function runJar(jarname: string) {
+    // Find .jar files recursively in each workspace and in all directories
+    // Run .jar with currently open file
+    const workspaceFolders = vscode.workspace.workspaceFolders?.map(folder => folder.uri) || [];
+    if (workspaceFolders.length == 0) {
+        vscode.window.showErrorMessage("No workspace folder opened!");
+    } else {
+        let foundJar;
+        for (const folder of workspaceFolders) {
+            foundJar = await findJarAndRun(folder, jarname);
+        }
+        if (!foundJar) {
+            vscode.window.showErrorMessage(`Unable to find ${jarname}.jar`)
+        }
+    }
+
+}
+
+async function findJarAndRun(uri: vscode.Uri, jarname: string) {
+    // Find terminal if exists
+    let terminal;
+    for (const term of vscode.window.terminals) {
+        const name = term.name;
+        if (name === jarname) {
+            terminal = term;
+        }
+    }
+    if (!terminal) {
+        terminal = vscode.window.createTerminal(jarname);
+    }
+
+    const jarUri = await findJar(uri, jarname);
+
+    if (jarUri) {
+        // Get filename of current file
+        const filepath = vscode.window.activeTextEditor?.document.uri.fsPath;
+        if (!filepath) {
+            terminal?.sendText(`java -jar ${jarUri.fsPath}`);
+        } else {
+            terminal?.sendText(`java -jar ${jarUri.fsPath} ${filepath}`);
```

Comment:
```suggestion
            // Get the extension of the current file
            const extension = filepath.split('.').pop();
            if (extension) {
                if (
                    (jarname === JSIM_LANGUAGE && extension !== JSIM_LANGUAGE) ||
                    (jarname === BSIM_LANGUAGE && extension !== 'uasm')
                ) {
                    vscode.window.showWarningMessage(
                        `You are running ${jarname}.jar with a source file with the .${extension} extension. Please be informed that this may not work properly.`,
                    );
                }
            }
            terminal?.sendText(`java -jar ${jarUri.fsPath} ${filepath}`);
```

The purpose of this part is to ensure that the user is informed that they are potentially running a jar file that is incompatible with the source code file that they are currently attempting to execute.